lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserSwiftDescriptorFinder.cpp

```diff
@@ -32,6 +32,29 @@ using namespace lldb_private;
 using namespace lldb_private::dwarf;
 using namespace lldb_private::plugin::dwarf;
 
+/// Given a die to a substituted generic Swift type, return the analogous
+/// unsubstituted version.
+///
+/// Given a generic type (for example, a generic Pair), the compiler will emit
+/// full debug information for the unsubstituted type (Pair<T, U>), and opaque
+/// debug information for each every specialization (for example, Pair<Int,
+/// Double>), with a link back to the unsubstituted type. When looking up one of
+/// the specialized generics, return the unsubstituted version instead.
+static std::optional<std::pair<CompilerType, DWARFDIE>>
+findUnsubstitutedGenericTypeAndDie(TypeSystemSwiftTypeRef &ts,
+                                   const DWARFDIE &die) {
+  auto specified_die =
+      die.GetAttributeValueAsReferenceDIE(llvm::dwarf::DW_AT_specification);
+  if (!specified_die)
+    return {};
+
+  const auto *mangled_name = specified_die.GetAttributeValueAsString(
+      llvm::dwarf::DW_AT_linkage_name, nullptr);
+  assert(mangled_name);
+  auto specified_type =
+      ts.GetTypeFromMangledTypename(ConstString(mangled_name));
+  return {{specified_type, specified_die}};
```

Comment:
only a single data point, but I would find it more clear if these variables were named `unsubstituted_` instead of `specified_`. Or even `generic_`. Or is `specified_` a widely used dwarf naming?

Reply:
Yeah the dwarf attribute is DW_AT_SPECIFICATION, but I agree that it makes more sense to call this unsubstituted in this context.